### PR TITLE
CLN: standardize different freq message

### PR DESCRIPTION
**Pull request by jbrockmendel**
In #24282 some of the PeriodArray/PeriodIndex methods couldn't accept the decorator because of mismatches between _DIFFERENT_FREQ and DIFFERENT_FREQ_INDEX.  This unifies the two.

The remaining case to figure out is what to render for `other_freq` when dealing with an ndarray.  Right now the placeholder is "FIXME".

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Perhaps hold off on this for #24024, which is changing (some) of these to use `_check_compatiable_with`.

I suppose that would be a relatively easy fix to split off from #24024, and it'd mean fewer changes here as well.

File: pandas/core/arrays/period.py [hunk 1]
```diff
@@ -765,10 +779,15 @@ def _add_delta(self, other):
         """
         if not isinstance(self.freq, Tick):
             # We cannot add timedelta-like to non-tick PeriodArray
-            raise IncompatibleFrequency("Input has different freq from "
-                                        "{cls}(freq={freqstr})"
-                                        .format(cls=type(self).__name__,
-                                                freqstr=self.freqstr))
+            if np.ndim(other) > 0:
+                # FIXME: What should this be rendered as?
+                other_freq = "FIXME"
```
Comment (TomAugspurger) [inline pandas/core/arrays/period.py]: Does "None" make sense here?

Comment (jbrockmendel) [reply pandas/core/arrays/period.py]: Either that or something resembling a least common denominator.  None sounds easier.

**Comment by jbrockmendel** [comment]
> I suppose that would be a relatively easy fix to split off from #24024, and it'd mean fewer changes here as well.

+1

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1545,9 +1547,8 @@ cdef int64_t[:] localize_dt64arr_to_period(int64_t[:] stamps,
     return result
 
 
-_DIFFERENT_FREQ = "Input has different freq={1} from Period(freq={0})"
-DIFFERENT_FREQ_INDEX = ("Input has different freq={1} "
-                        "from PeriodIndex(freq={0})")
+DIFFERENT_FREQ = ("Input has different freq={other_freq} "
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx]: why don't you just make this a function and have it raise ``IncompatileFrequencey`` as well?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/period.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx]: Sure.  I'm going to wait on this until after #24285 is merged.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @TomAugspurger if ok for you to merge ping (or merge)

File: pandas/core/arrays/period.py [hunk 3]
```diff
@@ -372,15 +372,13 @@ def __setitem__(
             value = period_array(value)
 
             if self.freqstr != value.freqstr:
```
Comment (TomAugspurger) [inline pandas/core/arrays/period.py]: `PeriodArray.__setitem__` is being removed, so I was hoping to not change these.

**Pull request merged**